### PR TITLE
Bug 1513527 - Switch from lodash to lodash-es

### PR DESCRIPTION
**Pull request by edmorley**
Since it allows webpack to tree shake unused modules.

### CODE REVIEW

**Comment by edmorley** [comment]
I believe the test failures here are due to quirks with karma-webpack, which should go away once we migrate fully to Jest.

**Comment by edmorley** [comment]
I've rebased and added a workaround for the karma-webpack issue, however the Jest tests are failing since lodash-es does not have an ES5 build alongside the ES6 one, and Jest doesn't yet know how to handle ES6 imports/exports. For our own code, Babel compiles these imports/exports down to CJS when using Jest, however Neutrino is currently not configured to transpile dependencies, so these are still ES6 when used by Jest. 

tl;dr this is blocked on facebook/jest#4842.

**Comment by edmorley** [comment]
Let's not bother with this for now, seeing as lodash 5 will resolve the issue more easily.
